packages/client/src/client.ts

```diff
@@ -23,6 +25,10 @@ interface Config<FetcherRequestInit extends RequestInit = RequestInit> {
   beforeRequest?: (
     fetchOptions?: FetcherRequestInit,
   ) => Promise<Partial<FetcherRequestInit> | undefined> | Partial<FetcherRequestInit> | undefined;
+  onError?: (
+    error: BigCommerceGQLError,
+    queryType: 'query' | 'mutation' | 'subscription',
+  ) => Promise<void> | void;
```

Comment:
Is this an existing pattern or something we built for this use case?

Reply:
It's built for this use-case. As @chanceaclark  was saying, we need the auth failure handling to be part of the DAL, which will require having this `onError` handler built into the client. 👍